core/src/main/java/org/elasticsearch/action/fieldstats/FieldStats.java

```diff
@@ -329,6 +329,10 @@ public final void writeTo(StreamOutput out) throws IOException {
                 writeMinMax(out);
             }
         } else {
+            if (hasMinMax == false) {
+                throw new IllegalArgumentException("cannot serialize null min/max fieldstats in a mixed-cluster " +
```

Comment:
this only happens if we mess something up right? I wonder if we should make it an assertion instead?